relative_to('/etc') PurePosixPath('passwd') >>> p. relative_to('/usr') Traceback (most recent call last): File "<stdin>", line 1, in <module> File "pathlib. py", line 941, in relative_to raise ValueError(error_message. format(str(self), str(formatted))) ValueError: '/etc/passwd' is not in the subpath of '/usr' OR one path is relative and the other is absolute. ValueError >>> p. relative_to('/usr', walk_up=True) PurePosixPath('. /etc/passwd') >>> p. relative_to('foo', walk_up=True) Traceback (most recent call last): File "<stdin>", line 1, in <module> File "pathlib. py", line 941, in relative_to raise ValueError(error_message. format(str(self), str(formatted))) ValueError: '/etc/passwd' is not on the same drive as 'foo' OR one path is relative and the other is absolute. PurePath resolve() walk_up=False name >>> p = PureWindowsPath('c:/Downloads/pathlib. tar. gz') >>> p. with_name('setup. py') PureWindowsPath('c:/Downloads/setup. py') >>> p = PureWindowsPath('c:/') >>> p. with_name('setup. py') Traceback (most recent call last): File "<stdin>", line 1, in <module> File "/home/antoine/cpython/default/Lib/pathlib. py", line 751, in with_name raise ValueError("%r has an empty name" % (self,)) ValueError: PureWindowsPath('c:/') has an empty name stem >>> p = PureWindowsPath('c:/Downloads/draft. txt') >>> p. with_stem('final') PureWindowsPath('c:/Downloads/final. txt') >>> p = PureWindowsPath('c:/Downloads/pathlib. tar. gz') >>> p. with_stem('lib') PureWindowsPath('c:/Downloads/lib. gz') >>> p = PureWindowsPath('c:/') >>> p. with_stem('') Traceback (most recent call last): File "<stdin>", line 1, in <module> File "/home/antoine/cpython/default/Lib/pathlib. py", line 861, in with_stem return self. with_name(stem + self.